lenskit/lenskit/algorithms/basic.py

```diff
@@ -80,6 +84,55 @@ def __str__(self):
         return "PopScore({})".format(self.score_method)
 
 
+class TimeBoundedPopScore(PopScore):
+    """
+    Score items by their time-bounded popularity, i.e., the popularity in the
+    most recent `time_window` period.  Use with :py:class:`TopN` to get a
+    most-popular-recent-items recommender.
+
+    Args:
+        time_window(datetime.timedelta):
+            The time window for computing popularity scores.
+        score_type(str):
+            The method for computing popularity scores.  Can be one of the following:
+
+            - ``'quantile'`` (the default)
+            - ``'rank'``
+            - ``'count'``
+
+    Attributes:
+        item_scores_(pandas.Series):
+            Time-bounded item popularity scores.
+    """
+
+    def __init__(self, time_window: timedelta, score_method="quantile"):
```

Comment:
Specifying this as a time delta introduces a behavioral dependency on the current time when the data is fit, which seems problematic for consistency and reproducibility. The same call with the same inputs will select different data based on when it is run.

I think it would probably be better to specify this as a cutoff time (which can then be set to predictably simulate a variety of times in an experiment).

(An alternative would be to specify it as a time delta that is computed from when the query is scored — queries will eventually get support for "query time" for offline experiments — instead from when the model is trained, but that would involve re-computing scores at query time which would be expensive).